ocaml/testsuite/tests/typing-modes/modes.ml

```diff
@@ -5,25 +5,25 @@
 (* Let bindings *)
 let local_ foo : string @@ unique = "hello"
 [%%expect{|
-Line 1, characters 4-43:
+Line 1, characters 36-43:
 1 | let local_ foo : string @@ unique = "hello"
-        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+                                        ^^^^^^^
```

Comment:
I think the original error message is better. `"hello"` is weakened to `local_`, which then doen't satisfy the requirement that top-level values must be `global`. Therefore, `This value` should include `local_`.

Reply:
I found the place that caused this regression, but I'm not quite sure I agree with you. The error should be underlining the value that escapes it's region, which means it should either underline `foo` or `"hello"`. In fact, running the compiler will underline `foo` (both in this branch and in the main branch), only the toplevel will underline differently like this. Apparently expect-tests are run in the toplevel.

Reply:
> The error should be underlining the value that escapes it's region

I agree with this, and I think there are several interpretaions. For a starter:
```
let g : 'a -> unit = fun _ -> ()
let f = g (local_ "hello")
```
Obviously we should point to `local_ "hello"` - because that's the expression trying to escape. `"hello"` is not trying to escape - it's simply going to `local_` who localizes the string successfully.

For let bindings in structures (not `let ... in ...`), it's more subtle:
```
let local_ foo = "hello"
```
I think I'm OK with pointing to either `local_ foo = "hello"`, for reasons similar to above, or pointing to `foo` only, for the following reasons:

When type checking a structure, we think we first collect a set of bindings with their types&modes, such as the `foo` above. Then, we try to put those bindings into a structure, during which we require the bindings to be `global`. In this sense, submode errors should point to either the whole definition `let local_ foo = "hello"`, or for simplicity only the name of the binding `foo`.

I agree that the top-level behaving differently than usual let-binding is bad and we should fix that (in a separate PR).